underscore.js

```diff
@@ -1239,6 +1246,12 @@
   // Shortcut function for checking if an object has a given property directly
   // on itself (in other words, not on a prototype).
   _.has = function(obj, key) {
+    if (_.isString(key) && key.indexOf('.') !== -1) {
+      var keys = key.split('.');
+      key = keys.pop();
+      var path = keys.join('.');
+      obj = resolveObject(obj, path);
+    }
```

Comment:
Ah, thanks for the input Nick! I'll look into when I get a chance. Thanks.

Sincerely,
Victor Bjelkholm

On Sat, Jan 3, 2015 at 5:14 PM, Nick Fisher <notifications@github.com>
wrote:

> In underscore.js
> <https://github.com/jashkenas/underscore/pull/1998#discussion-diff-22432397>
> :
>
> > @@ -1239,6 +1246,12 @@
> >    // Shortcut function for checking if an object has a given property directly
> >    // on itself (in other words, not on a prototype).
> >    _.has = function(obj, key) {
> > +    if (_.isString(key) && key.indexOf('.') !== -1) {
> > +      var keys = key.split('.');
> > +      key = keys.pop();
> > +      var path = keys.join('.');
> > +      obj = resolveObject(obj, path);
> > +    }
>
> I guess this would mean that you could never check for the existence of a
> key which contains a dot:
>
> _.has({ 'my.key': true}, 'my.key') === false
>
> You could make it more like:
>
> return obj != null && (hasOwnProperty.call(obj, key) || resolveObject(obj, path));
>
> *(There'd be an issue there since resolve object returns the value, not
> the bool showing existence of the key, but you get the idea.)*
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jashkenas/underscore/pull/1998/files#r22432397>.
>